libmamba/include/mamba/core/context_params.hpp

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2019, QuantStack and Mamba Contributors
+//
+// Distributed under the terms of the BSD 3-Clause License.
+//
+// The full license is in the file LICENSE, distributed with this software.
+
+#pragma once
+
+#include <string>
+
+#include "mamba/fs/filesystem.hpp"
+
+// TODO: having a file for this single structure is a bit of
```

Comment:
Side comment: the ideal setup would be each "service" or "module" (in the organisation meaning, not C++-specific) should expose in it's header the structs which serve as input to the whole "service". So structs would be located in their headers.
But here these two seems to be common for multiple parts of the code so OK.
However if there is a more general meaning for these I would prefer the name of the header to match that meaning as it's not really a context thing? Maybe `common_params.hpp`?

Reply:
Yeah I'm not sure yet of how things will be organized in the end. Extracting the context from the transaction is a rabbit hole, and I'm pretty sure things will change again. For now, I could have put structures in dedicated headers aside their "service" but that means a lot of new headers I'm not sure to keep in the end. And defining everything in the Context header (because all of these are more or less gathering of configurables from the context) before the whole refactoring is done would lead to a lot of circular includes.